defensics/tcp_data_handler.py

```diff
@@ -13,23 +14,46 @@ def __init__(self, proxy: CoapProxy, tcp_server: TCPServer):
         logging.debug('local proxy: ' + str(self.proxy) + 'global proxy: ' + str(proxy))
 
     def run(self):
-        rc = 1
-        while rc != 0:
-            rc = self.proxy.run()
-            logging.debug('Proxy retry')
-        logging.debug(str(self.proxy.device_iface) + str(self.proxy.req_char_iface) + str(self.proxy.rsp_char_iface))
         self.tcp_server.start()
+        skip = False
```

Comment:
Overall this whole procedure is way too complicated. Refactor it into smaller methods with descriptive names. You can create a simple state machine to clearly define states. Proper error handling, less nested chunks of code.